File: testdata/fixtures/clojure/CWE-352.clj
;; CWE-352: CSRF - POST without anti-forgery
(ns fixture.cwe352
  (:require [compojure.core :refer :all]))

;; ruleid: clojure-csrf
(POST "/transfer" req (str "transferred " (:params req)))

;; ruleid: clojure-csrf
(POST "/update" request (handle-update request))

;; ruleid: clojure-csrf
(POST "/delete" [] "deleted")

;; ok: clojure-csrf
(POST "/safe" req
  (if (= (get-in req [:headers "x-csrf-token"]) anti-forgery-token)
    "ok" "forbidden"))

;; ok: clojure-csrf
(POST "/safe2" [] 
  (wrap-anti-forgery handler))

File: testdata/fixtures/rust/CWE-352.rs
// CWE-352: CSRF - Actix POST without token
use actix_web::{post, HttpResponse};

// ruleid: rust-csrf
#[post("/transfer")]
async fn transfer() -> HttpResponse {
    HttpResponse::Ok().body("transferred")
}

// ruleid: rust-csrf
#[post("/update")]
async fn update() -> HttpResponse {
    HttpResponse::Ok().body("updated")
}

#[post("/safe")]
// ok: rust-csrf
async fn safe_transfer(req: actix_web::HttpRequest) -> HttpResponse {
    if req.headers().contains_key("X-CSRF-Token") {
        HttpResponse::Ok().body("ok")
    } else {
        HttpResponse::Forbidden().body("csrf missing")
    }
}

#[post("/safe2")]
// ok: rust-csrf
async fn safe2(req: actix_web::HttpRequest) -> HttpResponse {
    if req.headers().contains_key("X-CSRF-Token") {
        HttpResponse::Ok().body("ok")
    } else {
        HttpResponse::Forbidden().body("no")
    }
}

#[post("/safe3")]
// ok: rust-csrf
async fn safe3(req: actix_web::HttpRequest) -> HttpResponse {
    let _ = req.headers().get("X-CSRF-Token");
    let token = "x";
    HttpResponse::Ok().body(token)
}
